### PR TITLE
Contact Form

### PR DESCRIPTION
**Pull request by lucaslyl**
1. Allows users to fill in details for individuals which the User would like to easily retrieve so that they can continue a conversation with either via call or email.
2. Include fields for capturing details about a contact, phone number, email, address, salutations, and the department the individual belongs to
3. Email address Implementation

### CODE REVIEW

**Comment by jurgenwerk** [comment]
1. I ran the PR workflow, you can see checks that were performed, there seem to be some broken tests
2. Add some screenshots to your PR description which shows the changes made

File: packages/drafts-realm/contact-form.gts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import {
+  FieldContainer,
+  CardContainer,
+  IconButton,
+} from '@cardstack/boxel-ui/components';
+import { Sparkle } from '@cardstack/boxel-ui/icons';
+import { tracked } from '@glimmer/tracking';
+
+class Isolated extends Component<typeof ContactForm> {
+  @tracked hasTitleField =
```
Comment (jurgenwerk) [inline packages/drafts-realm/contact-form.gts]: Use a getter. tracked would be a good choice if you set something to it, but since it's a derived (computed) value it should be a `get hasTitleField() {}`

File: packages/drafts-realm/contact-form.gts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            <label>Department: </label>
+            <@fields.department />
+          </div>
+        </div>
+      </div>
+
+    </CardContainer>
+
+    <style>
+      @import url('https://fonts.googleapis.com/css2?family=Manrope:wght@200..800&family=Playfair+Display:ital,wght@0,400..900;1,400..900&family=Ubuntu+Sans:ital,wght@0,100..800;1,100..800&display=swap');
```
Comment (jurgenwerk) [inline packages/drafts-realm/contact-form.gts]: Why a new font?

File: packages/drafts-realm/contact-form.gts (hunk 1 above)
Comment (lucaslyl) [reply packages/drafts-realm/contact-form.gts]: 1. Change the @tracked with @get method:
image

File: packages/drafts-realm/contact-form.gts (hunk 2 above)
Comment (lucaslyl) [reply packages/drafts-realm/contact-form.gts]: Sorry about that, I had remove the new font.

image

File: packages/drafts-realm/contact-form.gts (hunk 1 above)
Comment (jurgenwerk) [reply packages/drafts-realm/contact-form.gts]: You don't have to show screenshots of code changes. They are clearly visible in the commits. With screenshots I meant showing the changes in application's interface (i.e. demonstrate the features of your card using screenshots or video)

File: packages/base/ethereum-address.gts [hunk 3]
```diff
@@ -79,9 +79,9 @@ export default class EthereumAddressField extends FieldDef {
   static [useIndexBasedKey]: never;
   static async [deserialize]<T extends BaseDefConstructor>(
     this: T,
-    address: any,
+    string: any,
```
Comment (habdelra) [inline packages/base/ethereum-address.gts]: Let’s not shadow typescript types. `string` is an actual type too.

Comment (habdelra) [reply packages/base/ethereum-address.gts]: Also `string: any` is super confusing. Why would a variable called string not have a string type?

Comment (habdelra) [reply packages/base/ethereum-address.gts]: Generally I don’t understand why this change was necessary.

File: packages/drafts-realm/Chess/c09f9bb3-df3f-4337-a6bd-cc4c96a5caa3.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
     "type": "card",
     "attributes": {
       "fen": "r3k2r/1bq2ppp/p1ppbn2/1p2p3/QP2P3/P1N2N2/2PPBPPP/R4RK1 b kq - 2 12",
-      "description": null,
```
Comment (habdelra) [inline packages/drafts-realm/Chess/c09f9bb3-df3f-4337-a6bd-cc4c96a5caa3.json]: Why this change?

**Review by habdelra — COMMENTED** [review]
I’m seeing lots of superfluous changes in data that don’t really seem to have anything to do with this PR.

File: packages/drafts-realm/ContactCard/1.json [hunk 5]
```diff
@@ -19,7 +19,6 @@
       },
       "guestNames": [
         "Liliana Evergreen",
-        "Marcellus Wilde",
```
Comment (habdelra) [inline packages/drafts-realm/ContactCard/1.json:22]: Why?

File: packages/drafts-realm/Map/20ad25ba-de18-40dc-979e-4603f62b5a2b.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
   "data": {
     "type": "card",
     "attributes": {
-      "address": "Livarska Ulica 10, Ljubljana, Slovenia",
+      "address": "Dk Impian",
```
Comment (habdelra) [inline packages/drafts-realm/Map/20ad25ba-de18-40dc-979e-4603f62b5a2b.json]: Why?

File: packages/drafts-realm/Vendor/2.json [hunk 7]
```diff
@@ -6,7 +6,7 @@
         "name": "Best Bakery",
         "description": null,
         "logoURL": "../invoice-assets/vendor-logo-best-bakery.png",
-        "email": "contact@bestbakery.com",
+        "email": "contact@bestbakery",
```
Comment (habdelra) [inline packages/drafts-realm/Vendor/2.json]: Why?

File: packages/drafts-realm/address-info.gts [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+import { eq } from '@cardstack/boxel-ui/helpers';
+import {
+  Country,
+  State,
+  City,
+} from 'https://cdn.jsdelivr.net/npm/country-state-city@3.2.1/+esm';
+
+// import govukCountryAndTerritoryAutocomplete from 'https://cdn.jsdelivr.net/npm/govuk-country-and-territory-autocomplete@1.0.2/+esm';
+// import openregisterPickerEngine from 'https://cdn.jsdelivr.net/npm/openregister-picker-engine@1.2.1/+esm';
+// import accessibleAutocomplete from 'https://cdn.jsdelivr.net/npm/accessible-autocomplete@3.0.0/+esm';
```
Comment (habdelra) [inline packages/drafts-realm/address-info.gts]: Let’s not include commented out code

File: packages/drafts-realm/user-email.gts [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+  field,
+  contains,
+  StringField,
+} from 'https://cardstack.com/base/card-api';
+import { Component } from 'https://cardstack.com/base/card-api';
+import { BoxelInput, FieldContainer } from '@cardstack/boxel-ui/components';
+import { TextInputValidator } from '../base/text-input-validator';
+
+function isValidEmail(email: string): boolean {
+  let emailRegex = new RegExp(
```
Comment (habdelra) [inline packages/drafts-realm/user-email.gts:18]: I’ve never seen an email regex done right in over 30 years of experience 😄. Email format validation is actually a really hard thing to do. Perhaps we should just check for really simple things, like an “@“ and a “.” separating the host from the TLD.
